### PR TITLE
neovim: remove unnecessary dependency pkg-config

### PR DESCRIPTION
**Pull request by dundargoc**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by dundargoc** [comment]
@carlocab wanna take a look?

**Comment by SMillerDev** [comment]
Please don't tag people an hour after making a review. Someone will volunteer their time eventually, no need to rush them.

File: Formula/n/neovim.rb [hunk 1]
```diff
@@ -71,7 +71,6 @@ class Neovim < Formula
 
   depends_on "cmake" => :build
   depends_on "lpeg" => :build # needed at runtime in 0.10.0
-  depends_on "pkg-config" => :build
```
Comment (SMillerDev) [inline Formula/n/neovim.rb:74]: Was it never required? Or was this recently changed?

Comment (dundargoc) [reply Formula/n/neovim.rb:74]: Changed somewhat recently

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**